relentless/simulate/lammps.py

```diff
@@ -0,0 +1,290 @@
+        elif not isinstance(V, TriclinicBox):
+            raise TypeError('LAMMPS boxes must be derived from TriclinicBox')
+
+        Lx = V.a[0]
+        Ly = V.b[1]
+        Lz = V.c[2]
+        xy = V.b[0]
+        xz = V.c[0]
+        yz = V.c[1]
+
+        lo = -0.5*np.array([Lx,Ly,Lz])
+        hi = lo + V.a + V.b + V.c
+
+        return np.array([lo[0],hi[0],lo[1],hi[1],lo[2],hi[2],xy,xz,yz])
+
+    def attach_potentials(self, sim):
+        # lammps requires r > 0
+        flags = sim.potentials.pair.r > 0
+        r = sim.potentials.pair.r[flags]
+        Nr = len(r)
+        if Nr == 1:
+            raise ValueError('LAMMPS requires at least two points in the tabulated potential.')
+
+        # check that all r are equally spaced
+        dr = r[1:]-r[:-1]
+        if not np.all(np.isclose(dr,dr[0])):
+            raise ValueError('LAMMPS requires equally spaced r in pair potentials.')
+
+        def pair_map(sim,pair):
+            # Map lammps type indexes as a pair, lowest type first
+            i,j = pair
+            id_i = sim.type_map[i]
+            id_j = sim.type_map[j]
+            if id_i > id_j:
+                id_i,id_j = id_j,id_i
+
+            return id_i,id_j
+
+        # write all potentials into a file
+        file_ = sim.directory.file('lammps_pair_table.dat')
+        with open(file_,'w') as fw:
+            fw.write('# LAMMPS tabulated pair potentials\n')
+            for i,j in sim.ensemble.pairs:
+                id_i,id_j = pair_map(sim,(i,j))
+                fw.write(('# pair ({i},{j})\n'
+                          '\n'
+                          'TABLE_{id_i}_{id_j}\n').format(i=i,
+                                                          j=j,
+                                                          id_i=id_i,
+                                                          id_j=id_j)
+                        )
+                fw.write('N {N} R {rmin} {rmax}\n\n'.format(N=Nr,
+                                                            rmin=r[0],
+                                                            rmax=r[-1]))
+
+                u = sim.potentials.pair.energy((i,j))[flags]
+                f = sim.potentials.pair.force((i,j))[flags]
+                for idx in range(Nr):
+                    fw.write('{idx} {r} {u} {f}\n'.format(idx=idx+1,r=r[idx],u=u[idx],f=f[idx]))
+
+        # process all lammps commands
+        cmds = ['neighbor {skin} multi'.format(skin=self.neighbor_buffer)]
+        cmds += ['pair_style table linear {N}'.format(N=Nr)]
+        for i,j in sim.ensemble.pairs:
+            # get lammps type indexes, lowest type first
+            id_i,id_j = pair_map(sim,(i,j))
+            cmds += ['pair_coeff {id_i} {id_j} {filename} TABLE_{id_i}_{id_j}'.format(id_i=id_i,id_j=id_j,filename=file_)]
+
+        return cmds
+
+class InitializeFromFile(Initialize):
+    def __init__(self, filename, neighbor_buffer, units='lj', atom_style='atomic'):
```

Comment:
Since `units` and `atom_style` are used by both `InitializeFromFile` and `InitializeRandomly`, you should move them into the base `Initialize` to reduce duplication.